lib/honeybadger/plugins/rails.rb

```diff
@@ -29,6 +29,17 @@ def render_exception(arg, exception)
         end
       end
 
+      class ErrorSubscriber
+        def self.report(exception, handled:, severity:, context: {}, source: nil)
+          return if source && ::Honeybadger.config[:'rails.subscriber_ignore_sources'].any? { |regex| regex.match?(source) }
+
+          tags = ['reporter:rails.error_subscriber', "severity:#{severity}", "handled:#{handled}"]
```

Comment:
The rationale for the other two:
- `handled`: `Rails.error.handle` will not throw an error, but will report it. Having `handled` helps you know this error was not a crash that affected a user, for example. (There's also a case for using this whenever a user manually calls `.notify`, tbh, but that's a wider effort.)
- `reporter` : so you know the specific mechanism that caught the error. Sure, you could dig through the stack trace, but that's more effort and could be a long way deep. It also helps with debugging sometimes ("this error was caught by the `at_exit` handler instead of the Rack middleware; something must be wrong"). (Also a wider discussion: making this more standard, e.g. `reporter:rack.middleware`.)

But I think it's fine to not want to commit until we're certain.

Reply:
Cool, thanks for the clarification. I think both of these things could be useful concepts, and potentially applicable to other areas we report errors in Rails. Could you create an issue for these so that we can track them long-term? The `handled` option could be really useful—I would like to make it clearer which errors were seen by a user.

I'll merge this commit in the meantime.

Reply:
Ah, you can hold on. Need to update the tests.

Reply:
On second thought, I decided to keep the `handled` and `source` options, since those come straight from Rails.